### DISCUSSION TITLE
ecu-b IP not found

### DISCUSSION THREAD

**Discussion by equinoxe12** [ksheumaker/homeassistant-apsystems_ecur#154] [Q&A]
Hi,
I have an ECU-b installed and it works perfectly (access from the Android app EMA APSytem)
It's connected to my box withe wifi so it has an ip assigned by DHCP, but it's impossible to find his ip so I can not configure Home Assistant! 
I tried to find it from the box or the "Advanced Port Scanner" application but I can't find the Ecu-b ip address, could you tell me how I can find it?

**Comment by HAEdwin** [comment]
You'll have to look for an Espressif Inc. or ESP node in for example your DHCP server. Then ping the IP-Address and pull the plug on the ECU to make sure you have the right node. I would strongly recommend assigning a fixed IP-address to the ECU.

**Reply by Candice1973** [reply]
I have just installed home assistant and downloaded the AP Systems Integration.  
This will be my second thing i would have installed after HACS (so imagine i am a child doing this)
I see the APSystems integration in my repository after reading  the prerequisites and i have come up with the same issue as equinoxe12.
I cant find the IP address for my ECU-B to fix the IP-address.
I looked in the APSystems settings page on my computer and on my phone and nothing :-(
Now i read the answer *look for an Espressif Inc. or ESP node in for example your DHCP server* but i do not understand how to do this.
Could someone point to a youtube video i can watch to find out how to find IP addresses please.

**Reply by HAEdwin** [reply]
You will need to log in into your router if that's your DHCP server. Best practice is to have an IP-range for "static" nodes like home assistant or the ECU-B and a dynamic (DHCP) range for roaming devices and such. It depends on the type of router how you have to assign a static IP-address to the ECU-B. Then you know for sure which IP-address it holds. So what kind of router do you have and search from there on YouTube.

**Comment by doug-last** [comment]
on "my" moden i cannot have static ip's and dhcp assign random ip's for the ecu each reboot, solved that doing this: https://github.com/ksheumaker/homeassistant-apsystems_ecur/discussions/137
in your case you might not need all that work, use the command i used to find the ip on the script: 
`arp-scan -gl -t 15000 -r 2 | grep -h "Beijing LT Honway" | cut -f 1`
it takes 30 seconds and not always get result so try a few times, this scan everything on the network, filter for the wifi manufacturar of ecu and show only the ip on screen.

**Reply by Tilperbr** [reply]
Doug, eu sou bem novato nessa area. Tenho o mesmo problema com meus dois ecu b. Eu nao faço a minima ideia de onde e como criar esse script e o q mais precisa ser inserido nele pra poder rodar. Consegue me ajudar com mais detalhes?

**Reply by HAEdwin** [reply]
@Tilperbr @doug-last Hi Tilperbr, Observe que você também pode acompanhar o desenvolvimento desta integração por meio de: https://github.com/HAEdwin/homeassistant-apsystems_ecu_reader

**Reply by Tilperbr** [reply]
Ty Edwin, I´m going to take a look at it

**Reply by doug-last** [reply]
@Tilperbr não tenho certeza de como poderia fazer para inserir 2 ECU.
Esse comando que passei foi a solução que usei para buscar ecu na rede local, inicialmente eu fiz um script em shell script e adicionei uma automação para rodá-lo, em outro momento só adicionei um comando watch na execução do container docker do homeassitant.
No seu caso, não sei o que acontece se ficar trocando aleatoriamente qual a ECU que está sendo lida após já estar configurado nessa integração.
O que eu tentaria seria alterar o shell script que postei [aqui](https://github.com/ksheumaker/homeassistant-apsystems_ecur/discussions/137#discussioncomment-4940925): adicionando algum comando para adicionar as ECU aleatoriamente no arquivo hosts, ou aleatoriamente ignorar alguma ecu encontrada, caso não esteja já acontecendo acidentalmente pela baixa confiabilidade da resposta do comando que usei para buscar a ecu.
E feito isso, ver o que acontece para ver se atende a necessidade.
O melhor seria remover tudo envolvendo esse script que mencionei e adicionar a feature de a integração buscar regularmente por ECU's na rede ao invés de ter que escolher uma via ip durante a configuração, mas me faltou tempo para tentar faze-la e submete-la.
